src/main/java/com/gorlah/kappabot/function/slashdot/MobileSlashdotTitleParser.java

```diff
@@ -0,0 +1,34 @@
+package com.gorlah.kappabot.function.slashdot;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.MatchResult;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class MobileSlashdotTitleParser {
+
+    private final Pattern mobileSlashdotArticleUrlPattern = Pattern.compile("https://m\\.slashdot\\.org/story/\\d+");
+
+    private final MobileSlashdotTitleFetcher mobileSlashdotTitleFetcher;
+
+    public boolean containsMobileSlashdotUrl(String message) {
+        return mobileSlashdotArticleUrlPattern.matcher(message).find();
+    }
+
+    public List<String> getTitles(String message) {
+        return mobileSlashdotArticleUrlPattern.matcher(message).results()
+                .parallel()
```

Comment:
I think this is a bit overkill here

Reply:
My thought process was that it would perform the title fetch in parallel which would help speed up the potentially slow lookup (in the case where multiple links are posted in the same message).  I did some testing and while it might speed it up to a degree, I didn't necessarily notice any performance increases between parallel vs sequential so I can go ahead and remove this.